tools/clang/lib/SPIRV/SpirvEmitter.cpp

```diff
@@ -6093,7 +6093,7 @@ SpirvInstruction *SpirvEmitter::turnIntoElementPtr(
 SpirvInstruction *SpirvEmitter::castToBool(SpirvInstruction *fromVal,
                                            QualType fromType,
                                            QualType toBoolType) {
-  if (isSameScalarOrVecType(fromType, toBoolType))
+  if (isSameScalarOrVecMatType(fromType, toBoolType))
```

Comment:
I think you can just call `isSameType` function.

Reply:
done